.github/workflows/build-xmtpd.yml

```diff
@@ -48,3 +50,24 @@ jobs:
           tags: ${{ steps.meta.outputs.tags }}
           labels: ${{ steps.meta.outputs.labels }}
           build-args: "GIT_COMMIT=${{ github.event_name != 'pull_request' && github.sha || github.event.pull_request.head.sha }}"
+
+  deploy:
+    name: Deploy new images to infra
+    runs-on: ubuntu-latest
+    needs: push_to_registry
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - name: Print Digest
+        id: print
+        run: echo "digest ${{ needs.push_to_registry.outputs.digest }}"
+      - name: Deploy (dev)
+        uses: xmtp-labs/terraform-deployer@v1
+        timeout-minutes: 45
+        with:
+          timeout: 45m
+          terraform-token: ${{ secrets.TERRAFORM_TOKEN }}
+          terraform-org: xmtp
+          terraform-workspace: testnet
+          variable-name: xmtpd_server_docker_image
+          variable-value: "ghcr.io/xmtp/xmtpd@${{ needs.push_to_registry.outputs.digest }}"
```

Comment:
I've never referenced something by `needs.something` in a GHA.

Reply:
it took me 5 or so iterations to get it right :)